vscode-test-runner/workspace-1/.joyride/src/integration_test/nrepl_test.cljs

```diff
@@ -0,0 +1,39 @@
+(ns integration-test.nrepl-test
+  (:require [cljs.test :refer [testing is]]
+            [integration-test.macros :refer [deftest-async]]
+            ["nrepl-client" :as nrepl-client]
+            [promesa.core :as p]
+            ["vscode" :as vscode]
+            ["util" :as util]))
+
+(deftest-async start-and-stop-nrepl-server
+  (testing "Starts and stops the nrepl server"
+    (p/let [port (vscode/commands.executeCommand "joyride.startNReplServer")
+            _ (is (number? port))
+            client (.connect nrepl-client  #js {:port port})
+            eval (util/promisify client.eval)]
+      (p/create (fn [resolve reject]
+                  (.on client "error"
+                       (fn [error]
+                         (js/console.error "Error connecting to nREPL server:" error)
+                         (reject error)))
+                  (.once client "connect"
+                         (fn []
+                           ;; TODO: Inside this p/let `println` stops working
```

Comment:
This is not about the nrepl server, I think. It's Joyride.

If I do `(println "foo")` outside the `p/let` it prints. Inside it doesn't print.

And if I make a failing test inside that `p/let` the `cljs.test` report doesn't print either.

Reply:
Is this running inside or outside SCI?

Reply:
And isn't the nREPL server also Joyride?

Reply:
ok, like I hinted at here, when you send a message to an nREPL server, the printing is now set to printing nREPL messages to the client, instead of the normal print-fn

https://github.com/BetterThanTomorrow/joyride/pull/145#discussion_r1064451244

Reply:
Ah, I see. I'll write a test for that as well.

Reply:
Yeah, I think just using js/console.log or so will probably be best for debugging in tests

Reply:
We could consider using `js/console.log` for the test runner reports as well.

Reply:
ok

Reply:
The way you can alter this is using:

```
(binding [*print-fn* js/console.log] ...)
```

Reply:
You can wrap it in a multi-arity function